### ISSUE TITLE
Loose version for ES15 preset

### ISSUE THREAD

**Issue by timrijkse** [Blocklevel/blue#53]
What we've discussed with Miento

**Comment by ThaNarie** [comment]
Great ticket description for outsiders :)

**Comment by jonaskuiler** [comment]
Agree, it's probably about this setting https://www.npmjs.com/package/babel-preset-es2015-loose which is going to be deprecated. I've also read about what it does and I can't see the advantages (http://www.2ality.com/2015/12/babel6-loose-mode.html?m=1).

Thanks for the reminder @timrijkse :)

I'm closing this issue

**Comment by mientjan** [comment]
Please reopen, your argument is invalid. loose is a standard option in the es2015 preset. 

 + generates smaller code
 + code is a lot cleaner and readable
 - will not follow the exact es2016 spec, typescript also does not do this. The reason is that its unnecessary. 

The goal for a transpiler should be to emit exactly what a programmer would type when implementing a class or namespace unaided by a tool. Something that loose:true represents.

I also believe ideas should be shut down in this fashion. Closing tickets even before a reply really does not help.

**Comment by jonaskuiler** [comment]
@mientjan You are right about my argument about the plugin being deprecated is invalid. That was all the information available for me at the time and I wasn't ware of the standard option so I just made an assumption, I closed the issue due to the lack of description.

I see that a lot of libraries are using loose mode but that doesn't convince me that we should use it for the use of this skeleton, most users will not know what loose mode does and I don't want users to rely on this behavior to make code cleaner and more readable whilst ignoring the spec.

Further more I don't really understand what you mean with "Will not follow the exact es2016 spec, typescript also does not do this. The reason is that its unnecessary.", I've investigated about the loose mode and like you said it's also available in TypeScript, but this doesn't mean that the standard es2015 mode is considered "unnecessary" right?

I see that a lot of projects are using loose mode but as far as I can see they're all libraries. The authors of libraries can not ignore  the performance gain and reduced build size of this option so that's probably why they are using it. However, Blue is not a library but a skeleton and I think it's best to just embrace the ES6 spec because I can imagine that there are small difference between normal mode and loose mode, and I think we will prevent unforeseen problems by users in the future by doing so.

Some sources:
- https://github.com/facebookincubator/create-react-app/issues/198
- http://www.2ality.com/2015/12/babel6-loose-mode.html?m=1
- https://github.com/vuejs-templates/webpack/blob/master/template/.babelrc
- https://developit.github.io/babel-legacy-docs//docs/advanced/loose/

**Comment by jonaskuiler** [comment]
Maybe @ThaNarie can also advise

**Comment by ThaNarie** [comment]
I think we have two questions/debates here:

1) Is loose mode _better_ for what we need?
2) If so, do we want to include it in the skeleton by default?

**loose mode**

First, let's start with the current loose mode configuration option, since most of the information is (partly) outdated at the moment.

Loose mode is a setting that individual plugins can choose to 'implement' and expose that as a configuration option. The default used `babel-preset-es2015` itself has a loose mode option that enables it for all the plugins in that preset:
https://github.com/babel/babel/tree/master/packages/babel-preset-es2015

**spec mode**

Besides the *loose* option, there is also a `spec` option, that sort of does the opposite. Plugins either provide a loose or a spec option, depending on how they implement the default version of their plugin. If the default version aligns to the specs, it will provide a loose option to be less strict. If the default mode is already loose, it will provide a spec option to adhere to the spec.

Examples:
https://babeljs.io/docs/plugins/transform-es2015-arrow-functions/
https://babeljs.io/docs/plugins/transform-class-properties/

**conclusion**

I think this new option is a trend in the direction that allow for better defaults, while giving the users an option to either go more loose or go more strict per plugin. Not enabling loose mode in the default preset, doesn't mean that all plugins are too strict.
Maybe in the future more plugins will transition to 'better' defaults.

Also, sometimes loose mode emits cleaner code, other times it increases performance by emitting more verbose/complicated code.

Given the two reasons above, I think it's important to know why you want to add loose mode, since you might not want to enable it for _all_ plugins.

**In skeleton**

I think there is indeed a difference between library and skeleton. Library will be used as-is, skeleton can be changed per project. I think most people will be used to the default preset, and if they changed it in previous projects they can easily do it here as well.
If we do plan to change some of all plugin settings, it should be clearly mentioned in the docs.
